include/d/actor/d_a_fganon.h

```diff
@@ -4,68 +4,150 @@
 #include "f_op/f_op_actor.h"
 #include "d/d_cc_d.h"
 #include "d/d_bg_s_acch.h"
-#include "SSystem/SComponent/c_phase.h"
+#include "d/d_particle.h"
 
+class camera_class;
+class mant_class;
+class daBoko_c;
+class daPy_py_c;
 class mDoExt_McaMorf;
 class mDoExt_brkAnm;
 class mDoExt_btkAnm;
 
-class fganon_class : public fopAc_ac_c {
+class fganon_class : public fopEn_enemy_c {
 public:
-    /* 0x290 */ f32 m290;
-    /* 0x294 */ u8 m294[0x2AC - 0x294];
-    /* 0x2AC */ request_of_phase_process_class m2AC;
-    /* 0x2B4 */ request_of_phase_process_class m2B4;
+    /* 0x2AC */ request_of_phase_process_class mPhs1;
+    /* 0x2B4 */ request_of_phase_process_class mPhs2;
     /* 0x2BC */ u8 m2BC;
     /* 0x2BD */ u8 m2BD;
-    /* 0x2BE */ u8 m2BE;
+    /* 0x2BE */ u8 mSwitchNo;
     /* 0x2BF */ u8 m2BF;
-    /* 0x2C0 */ mDoExt_McaMorf* m2C0;
-    /* 0x2C4 */ mDoExt_brkAnm* m2C4;
-    /* 0x2C8 */ J3DModel* m2C8;
-    /* 0x2CC */ mDoExt_brkAnm* m2CC;
-    /* 0x2D0 */ u8 m2D0[0x386 - 0x2D0];
+    /* 0x2C0 */ mDoExt_McaMorf* mpMorf;
+    /* 0x2C4 */ mDoExt_brkAnm* mpBrkAnm1;
+    /* 0x2C8 */ J3DModel* mpKenModel;
+    /* 0x2CC */ mDoExt_brkAnm* mpBrkAnm2;
+    /* 0x2D0 */ s8 m2D0;
+    /* 0x2D1 */ u8 m2D1[0x2D4 - 0x2D1];
+    /* 0x2D4 */ dKy_tevstr_c mTevStr;
+    /* 0x384 */ s16 m384;
     /* 0x386 */ s16 m386;
     /* 0x388 */ s16 m388;
-    /* 0x38A */ u8 m38A[0x3AE - 0x38A];
+    /* 0x38A */ s16 m38A;
+    /* 0x38C */ cXyz m38C;
+    /* 0x398 */ s16 m398;
+    /* 0x39A */ s16 m39A;
+    /* 0x39C */ f32 m39C;
+    /* 0x3A0 */ f32 m3A0;
+    /* 0x3A4 */ s16 m3A4[5];
     /* 0x3AE */ s16 m3AE;
-    /* 0x3B0 */ u8 m3B0[0x3B6 - 0x3B0];
+    /* 0x3B0 */ f32 m3B0;
+    /* 0x3B4 */ s16 m3B4;
     /* 0x3B6 */ s16 m3B6;
-    /* 0x3B8 */ u8 m3B8[0x408 - 0x3B8];
-    /* 0x408 */ u8 m408;
-    /* 0x409 */ u8 m409[0x40C - 0x409];
-    /* 0x40C */ dCcD_Sph m40C;
-    /* 0x538 */ dCcD_Sph m538;
-    /* 0x664 */ u8 m664[0x674 - 0x664];
-    /* 0x674 */ J3DModel* m674;
-    /* 0x678 */ mDoExt_btkAnm* m678;
-    /* 0x67C */ mDoExt_brkAnm* m67C;
-    /* 0x680 */ u8 m680[0x68C - 0x680];
-    /* 0x68C */ u8 m68C;
-    /* 0x68D */ u8 m68D[0x68F - 0x68D];
+    /* 0x3B8 */ s16 m3B8;
+    /* 0x3BA */ u8 m3BA[0x3BC - 0x3BA];
+    /* 0x3BC */ f32 m3BC;
+    /* 0x3C0 */ f32 m3C0;
+    /* 0x3C4 */ JPABaseEmitter* mEmitters1[2];
+    /* 0x3CC */ JPABaseEmitter* mEmitters2[2];
+    /* 0x3D4 */ JPABaseEmitter* mEmitters3[2];
+    /* 0x3DC */ u32 m3DC;
+    /* 0x3E0 */ cXyz m3E0;
+    /* 0x3EC */ cXyz m3EC;
+    /* 0x3F8 */ cXyz m3F8;
+    /* 0x404 */ f32 m404;
+    /* 0x408 */ s8 m408;
+    /* 0x409 */ s8 m409;
+    /* 0x40A */ s8 m40A;
+    /* 0x40B */ s8 m40B;
+    /* 0x40C */ dCcD_Sph mBallTgSph;
+    /* 0x538 */ dCcD_Sph mBallAtSph;
+    /* 0x664 */ cXyz m664;
+    /* 0x670 */ s8 m670;
+    /* 0x671 */ s8 m671;
+    /* 0x672 */ u8 m672;
+    /* 0x673 */ s8 m673;
+    /* 0x674 */ J3DModel* EnergySphereModel;
```

Comment:
```suggestion
    /* 0x674 */ J3DModel* mpEnergySphereModel;
```
https://github.com/zeldaret/tww/blob/main/docs/coding_guidelines.md#naming-style